.github/workflows/pre_release.yaml

```diff
@@ -0,0 +1,36 @@
+name: Canary Release
+
+on:
+  workflow_run:
+    workflows: ["Build and Test"]
+    branches: [main]
+    types: 
+    - completed
+
+jobs:
+  canary-release:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: ./tools/github-actions/setup-deps
+
+    - name: Generate Release Manifests
+      run: make generate-manifests IMAGE=envoyproxy/gateway-dev TAG=latest OUTPUT_DIR=release-artifacts
+
+    - name: Delete canary tag
+      uses: dev-drprasad/delete-tag-and-release@v0.2.0
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      with:
+        tag_name: canary
+
+    - name: Recreate canary tag and release
```

Comment:
why not use `softprops/action-gh-release`, with `prerelease:true` does it delete older files ?

Reply:
I will have a test.

Reply:
This can work, updated.